ClavaLaraApi/src-lara-clava/clava/clava/code/StatementDecomposer.js

```diff
@@ -1,209 +1,201 @@
-		
-		newStmts.push($newReturnStmt);
-		
-		return newStmts;
-	}
-
-	decomposeDeclStmt($stmt) {
-		
-		// declStmt can have one or more declarations
-		let $decls = $stmt.decls;
-		
-		let newStmts = [];
-		
-		// Create a separate stmt for each declaration
-		for(let $decl of $decls) {
-			
-			if(!$decl.instanceOf("vardecl")) {
-				debug("StatementDecomposer.decomposeDeclStmt: not implemented for decl of type " + $decl.joinPointType);
-				newStmts.push(ClavaJoinPoints.declStmt($decl));
-				continue;
-			}
-			
-			// If vardecl has init, decompose expression
-			if($decl.hasInit) {
-				let decomposeResult = this.decomposeExpr($decl.init);				
-				newStmts = newStmts.concat(decomposeResult.stmts);
-				$decl.init = decomposeResult.$resultExpr;
-			}
-
-			newStmts.push(ClavaJoinPoints.declStmt($decl));
-		}
-		
-		
-		return newStmts;
-	}
-	
-	/**
-	 * @return {DecomposeResult}
-	 */
-	decomposeExpr($expr) {
-		
-		if($expr.instanceOf("binaryOp")) {
-						
-			return this.decomposeBinaryOp($expr);
-		}
-		
-		if($expr.numChildren === 0) {
-		//if($expr.instanceOf("varref") || $expr.instanceOf("literal")) {
-			let stmts = [];
-			//let dec = new DecomposeResult(stmts, $expr);
-
-			return new DecomposeResult(stmts, $expr);
-		}
-
-		this._throwNotImplemented("expressions", $expr.joinPointType);	
-	}
-	
-	decomposeBinaryOp($binaryOp) {
-		
-		let kind = $binaryOp.kind;
-		
-		if(kind === "assign") {
-		
-			let stmts = [];
-			
-			// Get statements of right hand-side
-			let rightResult = this.decomposeExpr($binaryOp.right);
-			stmts = stmts.concat(rightResult.stmts);
-
-			// Add assignment
-			let $newAssign = ClavaJoinPoints.assign($binaryOp.left, rightResult.$resultExpr);
-			stmts.push(ClavaJoinPoints.exprStmt($newAssign));
-			
-			return new DecomposeResult(stmts, $binaryOp.left);			
-		} 
-		// TODO: Not taking into account += and other cases
-
-		// Apply decompose to both sides
-		let leftResult = this.decomposeExpr($binaryOp.left);
-
-		let rightResult = this.decomposeExpr($binaryOp.right);
-		
-		let stmts = [];
-		stmts = stmts.concat(leftResult.stmts);
-		stmts = stmts.concat(rightResult.stmts);
-		
-		// Create operation with result of decomposition
-		let $newExpr = ClavaJoinPoints.binaryOp($binaryOp.kind, leftResult.$resultExpr, rightResult.$resultExpr, $binaryOp.type);
-		
-		// Create declaration statement with result to new temporary variable
-		let tempVarname = this._newTempVarname();
-		let tempVarDecl = ClavaJoinPoints.varDecl(tempVarname, $newExpr);
-		stmts.push(tempVarDecl.stmt);
-
-		return new DecomposeResult(stmts, ClavaJoinPoints.varRefFromDecl(tempVarDecl));					
-
-
-		//this._throwNotImplemented("binary operators", kind);	
-	}
-	
-	
+  /**
+   * @param {string} [tempPrefix="decomp_"]
+   * @param {number} [startIndex=0]
+   */
+  constructor(tempPrefix = "decomp", startIndex = 0) {
+    this.tempPrefix = tempPrefix ?? "decomp_";
```

Comment:
Default values are not the same ("decomp" vs "decomp_"), is this on purpose?

Also, default parameters together with ?? seems redundant, maybe favor ??, since it also handles null?

Reply:
Good catch. The value was supposed to be the same.
Using the two syntaxes is somewhat redundant but I did it for a reason: I prefer the default parameter syntax because it yields better documentation, but to be defensive I wanted to deal with any cases that passed in null explicitly for some reason, although it should not happen.

Reply:
I agree default parameters yield better documentation